src/iso_alloc_sanity.c

```diff
@@ -313,4 +313,27 @@ INTERNAL_HIDDEN void *_iso_alloc_memcpy(void *restrict dest, const void *restric
 
     return __iso_memcpy(dest, src, n);
 }
+
+INTERNAL_HIDDEN void *__iso_memset(void * dest, int b, size_t n) {
+    char *p_dest = (char *) dest;
+
+    while(n--) {
+        *p_dest++ = b;
+    }
+
+    return dest;
+}
+
+INTERNAL_HIDDEN void *_iso_alloc_memset(void * dest, int b, size_t n) {
+    if(n > SMALLEST_CHUNK_SZ) {
+        iso_alloc_zone_t *zone = search_chunk_lookup_table(dest);
+        void *user_pages_start = UNMASK_USER_PTR(zone);
+
+        if(user_pages_start <= dest && (user_pages_start + ZONE_USER_SIZE) > dest && n > zone->chunk_size) {
+            LOG_AND_ABORT("Detected an out of bounds write memset: dest=0x%p (%d bytes) size=%d", dest, zone->chunk_size, n);
+        }
+    }
+
+    return __iso_memset(dest, b, n);
```

Comment:
Why not `__builtin_memset` instead of re-implementing it?

Reply:
it is purposely an unoptimised/hardly optimisable (by compiler) version, it is not for performance purpose. What do you think @struct ?

Reply:
I had trouble using `__builtin_memcpy` when I overloaded it. IIRC it resulted in a recursive call so using a hand rolled version here seems safest.

Reply:
We might then want to add a comment to explain this, and maybe a TODO to investigate, since some libc have highly-optimized `memcpy`/`memset` implementations.